src/CBT.Core/Internal/DefaultNuGetDownloader.cs

```diff
@@ -0,0 +1,83 @@
+﻿using Microsoft.Build.Framework;
+using System;
+using System.IO;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CBT.Core.Internal
+{
+    /// <summary>
+    /// A class that will download the latest version of NuGet from NuGet.org.
+    /// </summary>
+    internal static class DefaultNuGetDownloader
+    {
+        private static readonly Uri NuGetUrl = new Uri("https://dist.nuget.org/win-x86-commandline/latest/nuget.exe");
```

Comment:
I thought about this but the only other use case is to download the credential provider bundle which is a zip file and not multiple files.  I'll send out a pull request tomorrow for the implementation of this which downloads the credential provider bundle and unzips it.